src/main/java/de/retest/recheck/ui/descriptors/RootElementFilter.java

```diff
@@ -0,0 +1,96 @@
+package de.retest.recheck.ui.descriptors;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+
+import de.retest.recheck.ignore.Filter;
+import de.retest.recheck.ui.image.Screenshot;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class RootElementFilter {
```

Comment:
Maybe I didn't understand it correctly, but wasn't the initial idea have an `applyFilter` method in `Element` and `RootElement`, similar to `applyChanges`?

However, I believe I prefer this approach similar to `TestReportFilter` as this also removes complexity from the core classes.

Reply:
I initially implemented an `applyFilter` method in respective classes. However, @diba1013 also preferred this approach for the exact same reason you mentioned. `RootElementFilter` probably isn't the best name for a utility class. And I would rename it if you come up with a suggestion.

Reply:
Personally I'm fine with the name. It is similar to `TestReportFilter`, which starts with the top container. The only thing we could consider is to have a `SutStateFilter` instead.

Reply:
In the case of surili, we have to filter entire SUT states. This would also correspond to the `TestReportFilter`, i.e., the top container. We can still offer public methods for `RootElement`, `Element`, etc.

Reply:
We will offer public methods for `SutState` and `RootElement` for now.